apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -1,311 +1,487 @@
+                            node: child_nodes
+                                .into_iter()
+                                .filter_map(|child| match child {
+                                    TreeData::QueryPlan(node) => Some(Box::new(node)),
+                                    _ => None,
+                                })
+                                .next(),
+                        }),
+                    )),
+                })]
+            }
+            "subgraph" => {
+                vec![TreeData::Trace(self.find_ftv1_trace(span))]
+            }
+            "supergraph" => {
+                //Currently some data is in the supergraph span as we don't have the a request hook in plugin.
+                child_nodes.push(TreeData::Supergraph {
+                    http: self.extract_http_data(span),
+                    client_name: span
+                        .attributes
+                        .get(&CLIENT_NAME)
+                        .and_then(Self::extract_string),
+                    client_version: span
+                        .attributes
+                        .get(&CLIENT_VERSION)
+                        .and_then(Self::extract_string),
+                    operation_signature: span
+                        .attributes
+                        .get(&APOLLO_PRIVATE_OPERATION_SIGNATURE)
+                        .and_then(Self::extract_string)
+                        .unwrap_or_default(),
+                });
+                child_nodes
+            }
+            "request" => {
+                vec![TreeData::Request(
+                    self.extract_root_trace(span, child_nodes),
+                )]
+            }
+            _ => child_nodes,
+        })
     }
-}
-
-/// A [`SpanExporter`] that writes to [`Reporter`].
-///
-/// [`SpanExporter`]: super::SpanExporter
-/// [`Reporter`]: apollo_spaceport::Reporter
-#[derive(Debug)]
-#[allow(dead_code)]
-pub(crate) struct Exporter {
-    collector: String,
-    graph: Option<StudioGraph>,
-    reporter: tokio::sync::OnceCell<Reporter>,
-    normalized_queries: HashMap<String, String>,
-}
 
-impl Exporter {
-    /// Create a new apollo telemetry `Exporter`.
-    pub(crate) fn new(collector: String, graph: Option<StudioGraph>) -> Self {
-        Self {
-            collector,
-            graph,
-            reporter: tokio::sync::OnceCell::new(),
-            normalized_queries: HashMap::new(),
+    fn extract_string(v: &Value) -> Option<String> {
+        if let Value::String(v) = v {
+            Some(v.to_string())
+        } else {
+            None
         }
     }
-}
-
-/// Apollo Telemetry exporter's error
-#[derive(thiserror::Error, Debug)]
-#[error(transparent)]
-pub(crate) struct ApolloError(#[from] apollo_spaceport::ReporterError);
 
-impl From<std::io::Error> for ApolloError {
-    fn from(error: std::io::Error) -> Self {
-        ApolloError(error.into())
+    fn extract_i64(v: &Value) -> Option<i64> {
```

Comment:
Same https://docs.rs/serde_json_bytes/latest/serde_json_bytes/value/enum.Value.html#method.as_i64

Reply:
Unfortunately this an `open_telemetry::Value`, and no such method exists.